### PR TITLE
Added Azure Repos VCS

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/froggit-go/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `go fmt ./...` for formatting the code before submitting the pull request.
- [x] This feature is included on all supported VCS providers - GitHub, Bitbucket cloud, Bitbucket server, and GitLab.

---
- Added Azure Repos 
- Updated Go to 1.19

### CODE REVIEW

File: vcsutils/utils.go [hunk 1]
```diff
@@ -132,8 +133,113 @@ func safeCopy(targetFile *os.File, reader *tar.Reader) error {
...
+
+	// Create a destination file for unzipped content
+	destinationFile, err := os.OpenFile(filepath.Clean(fullFilePath), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, f.Mode())
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := destinationFile.Close(); err != nil {
+			return
+		}
```
Comment (eyalbe4) [inline vcsutils/utils.go:222]: This defer (and the other one inside this function) should return the errorm if there isn't already an error return by this function.

File: vcsutils/utils_test.go [hunk 2]
```diff
@@ -97,11 +96,40 @@ func TestDiscardResponseBody(t *testing.T) {
...
+	assert.Equal(t, false, DefaultIfNotNil(new(bool)))
+	arr := []int{3, 4}
+	assert.Equal(t, []int{3, 4}, DefaultIfNotNil(&arr))
+	assert.Equal(t, []int(nil), DefaultIfNotNil(new([]int)))
+}
+
+func TestUnzip(t *testing.T) {
+	destDir, err := os.MkdirTemp("", "")
+	assert.NoError(t, err)
+	defer os.RemoveAll(destDir)
```
Comment (eyalbe4) [inline vcsutils/utils_test.go:126]: We should assert the error returned by the `RemoveAll` function. There's another usage usage of `RemoveAll` in this function that requires this fix as well.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
Let's also update the README documentation.

File: vcsutils/utils_test.go (hunk 2 above)
Comment (omerzi) [reply vcsutils/utils_test.go]: done

File: vcsutils/utils.go (hunk 1 above)
Comment (omerzi) [reply vcsutils/utils.go:223]: done

**Review by yahavi — CHANGES_REQUESTED** [review]
Let's also add `vcsutils.AzureRepos` in TestClientBuilder.

File: vcsclient/azurerepos.go [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+
+func (client *AzureReposClient) buildAzureReposClient(ctx context.Context) error {
+	baseUrl := strings.TrimSuffix(client.vcsInfo.APIEndpoint, string(os.PathSeparator))
+	client.connectionDetails = azuredevops.NewPatConnection(baseUrl, client.vcsInfo.Token)
+	azureGitClient, err := git.NewClient(ctx, client.connectionDetails)
+	client.azureReposGitClient = &azureGitClient
+	return err
+}
+
+func (client *AzureReposClient) TestConnection(ctx context.Context) error {
```
[the remark this answers is not in the document]
Comment (omerzi) [reply vcsclient/azurerepos.go:38]: done

File: vcsclient/azurerepos.go [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+	if req, err = http.NewRequestWithContext(ctx, http.MethodGet, downloadRepoUrl, nil); err != nil {
+		return
+	}
+	for key, val := range headers {
+		req.Header.Add(key, val)
+	}
+	if res, err = httpClient.Do(req); err != nil {
+		return
+	}
+	if statusOk := res.StatusCode >= 200 && res.StatusCode < 300; !statusOk {
```
[the remark this answers is not in the document]
Comment (omerzi) [reply vcsclient/azurerepos.go:134]: done

File: vcsclient/azurerepos_test.go [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+	jsonRes, err := json.Marshal(res)
+	assert.NoError(t, err)
+	ctx := context.Background()
+	client, cleanUp := createServerAndClient(t, vcsutils.AzureRepos, true, jsonRes, "listRepositories", createAzureReposHandler)
+	defer cleanUp()
+	reposMap, err := client.ListRepositories(ctx)
+	assert.NoError(t, err)
+	for _, repos := range reposMap {
+		assert.ElementsMatch(t, repos, testRepos)
+	}
```
[the remark this answers is not in the document]
Comment (omerzi) [reply vcsclient/azurerepos_test.go:50]: done

File: vcsclient/azurereposcommon.go [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package vcsclient
```
[the remark this answers is not in the document]
Comment (omerzi) [reply vcsclient/azurereposcommon.go:1]: done

File: vcsutils/utils.go [hunk 7]
```diff
@@ -132,8 +133,115 @@ func safeCopy(targetFile *os.File, reader *tar.Reader) error {
...
+	}()
+
+	for {
+		if _, err = io.CopyN(destinationFile, zippedFile, 1024); err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			return err
+		}
+	}
```
[the remark this answers is not in the document]
Comment (omerzi) [reply vcsutils/utils.go:245]: I changed the zip/tar parameter in safeCopy to inteface { io.Reader}, what do you think?

[the remark this answers is not in the document]
Comment (yahavi) [reply vcsutils/utils.go:245]: Sounds good to me

File: vcsclient/azurerepos_test.go [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+	client, cleanUp := createServerAndClient(
+		t,
+		vcsutils.AzureRepos,
+		true,
+		repoFile,
+		fmt.Sprintf("/%s/_apis/git/repositories/%s/items/items?path=/&[…]ptor[version]=%s&$format=zip",
+			"",
+			repo1,
+			branch1),
+		createAzureReposHandler)
```
[the remark this answers is not in the document]
Comment (omerzi) [reply vcsclient/azurerepos_test.go:91]: done

**Pull request merged**